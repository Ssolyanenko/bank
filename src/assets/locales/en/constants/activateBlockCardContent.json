{
  "activateBlockCardContent": {
    "blocked": {
      "text": "Are you sure you want to block your card?",
      "buttonName": "Yes, block it",
      "terms": "Blocking the card will restrict all and any operations with it, but all card-related service fees will still be charged."
    },
    "activate": {
      "text": "Are you sure you want to activate your card?",
      "buttonName": "Yes, activate it",
      "terms": "If you activate the card, you will be able to use it for all kinds of transactions again. Reactivation is free of charge, all card-related service fees will remain the same. Would you like to activate your card?"
    },
    "status": "Change card status"
  }
}
