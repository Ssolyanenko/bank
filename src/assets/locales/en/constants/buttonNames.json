{
  "buttonNames": {
    "orderCard": "Order a card",
    "back": "Back",
    "send": "Send",
    "logIn": "Log in",
    "logOut": "Log out",
    "forgotPassword": "Forgot password?",
    "learnMore": "Learn more",
    "setPinCode": "Set pin code",
    "continue": "Continue",
    "activate": "Activate",
    "ok": "OK",
    "submit": "Submit",
    "cancel": "Cancel",
    "hideFilters": "Hide filters",
    "resetAllFilters": "Reset all filters",
    "apply": "Apply",
    "sendAgain": "Send again",
    "save": "Save",
    "changePinCode": "Change pin code",
    "change": "Change",
    "cardStatement": "Card statement ",
    "showStatement": "Show statement",
    "sendToEmail": "Send to email",
    "downloadStatement": "Download statement",
    "shareStatement": "Share statement",
    "cardDetails": "Card Details",
    "reset": "Reset",
    "showReceipt": "Show receipt",
    "downloadReceipt": "Download receipt",
    "shareReceipt": "Share receipt",
    "leave": "Leave",
    "goToMainPage": "Go to main page",
    "confirm": "Confirm",
    "goToMyCards": "Go to My Cards",
    "contactBank": "Contact the bank"
  }
}
