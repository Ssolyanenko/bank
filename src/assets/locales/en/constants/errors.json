{
  "errors": {
    "required": "The field is required to fill out",
    "terms": "Please, accept the terms and conditions and confirm the accuracy of information provided.",
    "invalid": "Invalid {{name}} format, please try again",
    "min": "Required number of characters is {{min}}",
    "max": "Maximum number of characters is {{max}}",
    "minCardAmount": "The minimum amount of a credit is £{{min}}",
    "maxAmount": "The maximum amount of a credit is £{{max}}",
    "allFields": "You haven\"t filled out all required fields",
    "limitChanges": "The limit amount can be from 0.00 to 25,000.00 currency units",
    "somethingWentWrong": "Something went wrong. Please try again.",
    "doesNotMatch": "Your pin codes do not match",
    "noBanksFound": "Sorry, no banks are found, change the city",
    "onlyDigits": "Only digits are allowed",
    "compulsory": "This field is compulsory",
    "minCategories": "You must select 3 categories",
    "maxCategories": "Only 3 categories allowed",
    "chooseCard": "Before entering the amount, please, choose a card",
    "minAmount": "The minimum transfer amount is 1 GBP",
    "minEmail": "The e-mail must be at least 5 characters long",
    "maxEmail": "The maximum number of characters is {{max}}",
    "passwordValidation": "Field should contain letters (at least 1 capital and 1 lowercase letter) and numbers",
    "loseData": "If you leave now, the data you entered will be lost. Are you sure you want to leave?"
  }
}
