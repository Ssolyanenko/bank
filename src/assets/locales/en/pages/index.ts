import landingPage from './landingPage.json';
import mainPageContent from './mainPageContent.json';
import notFoundPage from './notFoundPage.json';
import myCardsPage from './myCardsPage.json';

export const translationsPages = {
  ...landingPage,
  ...mainPageContent,
  ...notFoundPage,
  ...myCardsPage,
};
