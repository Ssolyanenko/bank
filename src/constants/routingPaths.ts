export enum RoutingPaths {
  APPLY_LOAN = 'apply-loan',
  CARDS = 'cards',
  CARD_PRODUCTS = 'card-products',
  CARDS_APPLICATIONS = 'card-applications',
  CREDIT_CARDS = 'credit-cards',
  CREDIT_CARD_BILLABLE = 'billable',
  CREDIT_CARD_BILLABLE_PREMIUM = 'billable-premium',
  DEALS_OFFERS = 'deals-offers',
  DEBIT_CARDS = 'debit-cards',
  DEBIT_SMART_CARD = 'debit-smart-card',
  DEBIT_TRAVEL_CARD = 'debit-travel-card',
  DEPOSITS = 'deposits',
  INVESTING = 'investing',
  INSURANCE = 'insurance',
  MAIN_PAGE_URL = 'main-page',
  MY_CARDS = 'my-cards',
  MY_LOANS = 'my-loans',
  LOAN_PRODUCTS = 'loan-products',
  LOANS = 'loans',
  TRANSFERS = 'transfers',
  USER_ACCOUNT = 'user-account',
  CARD_DETAILS = 'card-details',
  PAGE_NOT_FOUND = 'page-not-found',
  CONTACTS = 'contacts',
}
