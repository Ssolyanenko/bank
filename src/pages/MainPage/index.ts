export * from './MainPage';
export * from './MainPageContent';
