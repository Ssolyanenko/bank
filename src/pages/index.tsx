export * from 'pages/CreditCardsPage';
export * from 'pages/LandingPage';
export * from 'pages/MainPage';
export * from 'components/_basic/SecureWrapper';
export * from 'pages/MyCardApplicationsPage';
export * from 'pages/MyCardsPage';
export * from 'pages/CreditCardPremiumPage';
export * from 'pages/CreditCardBillablePage';
export * from 'pages/LayoutPage';
export * from 'pages/NotFoundPage';
