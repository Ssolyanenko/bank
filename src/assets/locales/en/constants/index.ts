import menu from './menu.json';
import errors from './errors.json';
import transferListContent from './transferListContent.json';
import stepForm from './stepForm.json';
import buttonNames from './buttonNames.json';
import lockCardModalContent from './lockCardModalContent.json';
import user from './user.json';
import input from './input.json';
import activateBlockCardContent from './activateBlockCardContent.json';
import resetFormsModalsContent from './resetFormsModalsContent.json';

export const translationsConstants = {
  ...menu,
  ...errors,
  ...transferListContent,
  ...stepForm,
  ...buttonNames,
  ...lockCardModalContent,
  ...user,
  ...input,
  ...activateBlockCardContent,
  ...resetFormsModalsContent,
};
