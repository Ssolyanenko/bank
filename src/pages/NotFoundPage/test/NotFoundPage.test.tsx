import 'jsdom-global/register';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';

import { NotFoundPage } from 'pages';
import { RoutingPaths } from 'constants/routingPaths';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: (): ((path: string, options?: { replace?: boolean }) => void) => mockedUsedNavigate,
}));

describe('<NotFoundPage />', (): void => {
  test('should make a snapshot', (): void => {
    const wrapper = shallow(
      <MemoryRouter>
        <NotFoundPage />
      </MemoryRouter>
    );
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should contain title block', (): void => {
    const wrapper = shallow(
      <MemoryRouter>
        <NotFoundPage />
      </MemoryRouter>
    );

    expect(wrapper.render().find('.title').text()).toEqual('Page not found');
  });

  test('should contain message block', (): void => {
    const wrapper = shallow(
      <MemoryRouter>
        <NotFoundPage />
      </MemoryRouter>
    );

    expect(wrapper.render().find('.message').text()).toEqual(
      'The page you are looking for could have been deleted or never existed.'
    );
  });

  test('should contain primary button', (): void => {
    const wrapper = shallow(
      <MemoryRouter>
        <NotFoundPage />
      </MemoryRouter>
    );

    expect(wrapper.render().find('.button').attr('name')).toEqual('primaryButton');
  });

  test('should call callback on click', (): void => {
    const wrapper = mount(
      <MemoryRouter>
        <NotFoundPage />
      </MemoryRouter>
    );

    const button = wrapper.find('button');
    button.simulate('click');
    expect(mockedUsedNavigate).toHaveBeenCalledWith(`/${RoutingPaths.MAIN_PAGE_URL}`);
  });

  test('should contain page not found image', (): void => {
    const wrapper = shallow(
      <MemoryRouter>
        <NotFoundPage />
      </MemoryRouter>
    );

    expect(wrapper.render().find('img').attr('alt')).toEqual('Page not found error');
  });
});
