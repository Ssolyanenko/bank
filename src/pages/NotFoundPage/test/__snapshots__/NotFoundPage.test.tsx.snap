// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should make a snapshot 1`] = `
<Router
  location={
    {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    {
      "action": "POP",
      "createHref": [Function],
      "encodeLocation": [Function],
      "go": [Function],
      "index": 0,
      "listen": [Function],
      "location": {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <NotFoundPage />
</Router>
`;
